docs/content/filtering.md

```diff
@@ -5,31 +5,65 @@ description: "Filtering, includes and excludes"
+to.
 
-The filters are applied for the `copy`, `sync`, `move`, `ls`, `lsl`,
-`md5sum`, `sha1sum`, `size`, `delete` and `check` operations.
-Note that `purge` does not obey the filters.
+They are specified in terms of path/file name
+patterns; path/file lists; file age and size, or presence of
+a file in a directory. Bucket remotes without the concept of
+directory apply filters to object key, age and size in an
+analogous way.
 
-Each path as it passes through rclone is matched against the include
-and exclude rules like `--include`, `--exclude`, `--include-from`,
-`--exclude-from`, `--filter`, or `--filter-from`. The simplest way to
-try them out is using the `ls` command, or `--dry-run` together with
-`-v`. `--filter-from`, `--exclude-from`, `--include-from`, `--files-from`,
-`--files-from-raw` understand `-` as a file name to mean read from standard
-input.
+Rclone `purge` does not obey filters.
 
-## Patterns ##
+To test filters without risk of damage to data, apply them to `rclone
+ls`, or with the `--dry-run` and `-vv` flags.
 
-The patterns used to match files for inclusion or exclusion are based
-on "file globs" as used by the unix shell.
+Rclone filter patterns can only be used in filter command line options, not
+in the specification of a remote.
 
-If the pattern starts with a `/` then it only matches at the top level
-of the directory tree, **relative to the root of the remote** (not
-necessarily the root of the local drive). If it doesn't start with `/`
-then it is matched starting at the **end of the path**, but it will
-only match a complete path element:
+Eg `rclone copy "remote:dir*.jpg" /path/to/dir` does not have a filter effect.
+`rclone copy remote:dir /path/to/dir --include "*.jpg"` does.
+
+**Important** Avoid mixing any two of `--include...`, `--exclude...` or
+`--filter...` flags in an rclone command. The results may not be what
+you expect. Instead use a `--filter...` flag.
+
+## Patterns for matching path/file names ##
+
+### Pattern syntax ###
+
+Rclone's matching rules follow a glob style:
+
+    `*`         matches any sequence of non-separator (`/`) characters
+    `**`        matches any sequence of characters
+    `?`         matches any single non-separator (`/`) character
+    `[` [ `!` ] { character-range } `]`
+                character class (must be non-empty)
+    `{` pattern-list `}`
+                pattern alternatives
+    c           matches character c (c != `*`, `**`, `?`, `\`, `[`, `{`, `}`)
+    `\` c       matches character c
+
+character-range:
+
+    c           matches character c (c != `\\`, `-`, `]`)
+    `\` c       matches character c
+    lo `-` hi   matches character c for lo <= c <= hi
+
+pattern-list:
+
+    pattern { `,` pattern }
+                comma-separated (without spaces) patterns
+
+If the filter pattern starts with a `/` then it only matches
+at the top level of the directory tree,
+**relative to the root of the remote** (not necessarily the root
```

Comment:
I'll just note that the "relative to the root of the remote" thing is something new users get wrong a lot.

Maybe it might be worth adding a common mistakes section?

- Not using paths relative to the root of the remote
- Not using `/` to match things at the root of the remote
- Not using `**` to match things in a directory

Reply:
Yes, I will try to have a go at that. The forum is stuffed full of folks missing** off an include or getting the roots wrong.